e2e/test/scenarios/dashboard-filters/reproductions/25322-loading-list-values.cy.spec.js

```diff
@@ -44,35 +49,19 @@ const createDashboard = () => {
     .then(({ body: { id: card_id } }) => {
       cy.createDashboard(dashboardDetails).then(
         ({ body: { id: dashboard_id } }) => {
-          cy.request("POST", `/api/dashboard/${dashboard_id}/cards`, {
-            cardId: card_id,
-            row: 0,
-            col: 0,
-            size_x: 4,
-            size_y: 4,
-          }).then(({ body: { id: dashcard_id } }) => {
-            cy.request("PUT", `/api/dashboard/${dashboard_id}/cards`, {
-              cards: [
+          addCardToDashboard({
+            dashboard_id,
+            card_id,
+            card: {
+              parameter_mappings: [
                 {
-                  id: dashcard_id,
                   card_id,
-                  row: 0,
-                  col: 0,
-                  size_x: 4,
-                  size_y: 4,
-                  parameter_mappings: [
-                    {
-                      card_id,
-                      parameter_id: parameterDetails.id,
-                      target: ["dimension", ["field", ORDERS.STATE, null]],
-                    },
-                  ],
+                  parameter_id: parameterDetails.id,
+                  target: ["dimension", ["field", ORDERS.STATE, null]],
                 },
               ],
-            }).then(() => {
-              return { dashboard_id };
-            });
-          });
+            },
+          }).then(() => ({ dashboard_id }));
```

Comment:
I don't believe this is what the test case was trying to test. The original [issue](https://github.com/metabase/metabase/issues/25322) (and assertion in this test) was that there should be a loading spinner while filter values are being loaded. This should happen regardless of if the filter was added to the dashcard during its initial creation or later.

It was a common pattern throughout our e2e tests to manually send a `POST` request to create the dashcard, then a `PUT` request to link a filter to it. Why was this the case? I'm not entirely sure, but I would assume it had to do with some limitation of the old `POST` endpoint. I read through the test cases and none of them seemed to be explicitly testing first creating then later adding the filter to the dashcard.

For this reason, I think it's safe to change it to doing it in the single `PUT` request. All the test cases still pass, the code is cleaner, and performance should be better since we are not making an additional roundtrip to the server.